test/e2e-tests/tests/pipelines_test.go

```diff
@@ -81,98 +51,219 @@ func init() {
+		t.Fatal(err.Error())
+	}
+
+	err = WaitForAllPipelineRunsToComplete(ctx, config)
 	if err != nil {
 		t.Fatal(err.Error())
 	}
 }
 
-func Test_GitOpsPipelineRunsComplete(t *testing.T) {
+func Test_GitOpsUpdatePipeline(t *testing.T) {
 	ctx := CreateContext()
 
-	options, err := support.GetOptions()
+	config, err := support.GetConfig()
 	if err != nil {
-		t.Fatal(err.Error())
+		panic(err.Error())
+	}
+
+	if !config.GitOpsConfig.Enabled {
+		t.Skipf("skipping %v, Gitops pipeline not configured", t.Name())
+	}
+
+	if !config.GitFetchConfig.Enabled && !config.S3FetchConfig.Enabled {
+		t.Skipf("skipping %v, neither git_fetch or s3_fetch was not enabled", t.Name())
 	}
 
-	clients, err := support.CreateClients(options.ClusterNamespace)
+	clients, err := support.CreateClients(config.Namespace)
 	if err != nil {
 		t.Fatal(err.Error())
 	}
 
-	pipelineRunPaths := []string{GitOpsUpdateTensorflowHousingPipelineRunRelativePath, GitOpsUpdateBikeRentalsPipelineRunRelativePath}
-
-	gitURL, err := support.ParseGitURL(options.GitRepo)
+	gitURL, err := support.ParseGitURL(config.GitOpsConfig.Repo)
 	if err != nil {
 		t.Fatal(err.Error())
 	}
 
-	for _, path := range pipelineRunPaths {
-		pipelineRun, err := support.ReadFileAsPipelineRun(path)
+	pipelineRunFilesAndLabels := []struct {
+		pipelineRun string
+		label       string
+	}{
+		{pipelineRun: GitOpsUpdateTensorflowHousingPipelineRunRelativePath, label: "model-name=tensorflow-housing"},
+		{pipelineRun: GitOpsUpdateBikeRentalsPipelineRunRelativePath, label: "model-name=bike-rentals-auto-ml"},
+	}
+
+	for _, p := range pipelineRunFilesAndLabels {
+		pipelineRun, err := support.ReadFileAsPipelineRun(p.pipelineRun)
 		if err != nil {
 			t.Fatal(err.Error())
 		}
 
 		support.SetPipelineRunParam("gitServer", support.NewStringParamValue(gitURL.Server), &pipelineRun)
-		support.SetPipelineRunParam("gitApiServer", support.NewStringParamValue(options.GitAPIServer), &pipelineRun)
+		support.SetPipelineRunParam("gitApiServer", support.NewStringParamValue(config.GitOpsConfig.ApiServer), &pipelineRun)
 		support.SetPipelineRunParam("gitOrgName", support.NewStringParamValue(gitURL.OrgName), &pipelineRun)
 		support.SetPipelineRunParam("gitRepoName", support.NewStringParamValue(gitURL.RepoName), &pipelineRun)
-		support.SetPipelineRunParam("gitRepoBranchBase", support.NewStringParamValue(options.GitBranch), &pipelineRun)
+		support.SetPipelineRunParam("gitRepoBranchBase", support.NewStringParamValue(config.GitOpsConfig.Branch), &pipelineRun)
+
+		// we need to get the results from the pipeline run created by the ml ops pipeline, we also need
+		// to get the correct one that is for this model. Using the correct label and ensuring it was
+		// created after the tests have started then we can ensure we are getting the correct one
+		completedPipelineRuns, err := config.Clients.PipelineRun.List(ctx, metav1.ListOptions{
+			LabelSelector: p.label,
+		})
+		if err != nil {
+			t.Fatal(err.Error())
+		}
```

Comment:
I think this and the timestamp check should be done first for performance reasons, only if we have a previous MLOps pipeline run should we continue with setting up a GitOps pipeline run. Wdyt?

Also, I think that in case a pipeline is not found, we can just call t.Fail() to not fail the test immediately and then call continue? But ideally this should be in separate tests probably. We can improve it later I think, it is just my observation now. I noticed you tried to fix it in #271 though it won't fail test if one MLOps pipeline fails. I know that the corresponding MLOps test fails before, so the result won't be "PASS" but I think we should still fail the GitOps test because it wasn't able to run properly for both MLOps pipelines. But as I said, this can be handled in that other PR.

Reply:
Updated now, they are separated and when looking for that pipeline run if it is failed then the gitops part will fail also now